naga/tests/in/wgsl/f16.wgsl

```diff
@@ -61,6 +61,14 @@ fn f16_function(x: f16) -> f16 {
    // Constructing a vec3<i64> from a i64
    val += vec3<f16>(input_uniform.val_f16).z;
 
+   // Cast f16 literals to other types
+   output.val_i32 = i32(65504h);
+   output.val_i32 = i32(-65504h);
+   output.val_u32 = u32(65504h);
+   output.val_u32 = u32(-65504h);
+   output.val_f32 = f32(65504h);
+   output.val_f32 = f32(-65504h);
```

Comment:
note: To myself: This value should set all significand bits to 1.

suggestion: Maybe a quick comment on this? 😀

question: Is there any other rationale or explanation for these values that we might usefully add here to this test? Maybe "cross product of min and max values converted to every other scalar type"?

Reply:
It’s chose this more because it’s the max and min finite number, rather than all significand bits being 1 (though of course those two things aren’t just coincidental). Will add a comment.

Converting any negative to unsigned was the only existing bug here, but corner cases such as maxes and mins often make for useful tests. Not really got any more rationale than that